### ISSUE TITLE
state/event/mass_transfer_case output columns not matching S1, S2 states

### ISSUE THREAD

**Issue by celiotine** [POSYDON-code/POSYDON#214]
There is a mismatch in the output between the `RLO`, `CC`, and `mass_transfer_case` values and the `S1_state`, `S2_state` values for some binaries, e.g. `CC2` happens for `S1`, etc. Here is an example of this behavior:
<img width="646" alt="Screen Shot 2023-12-07 at 2 56 59 PM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/ecc9be3c-c9e5-431b-a9fb-898835205a68">

For the `mass_transfer_case` strings, there are unique bugs where these strings are set in `cumulative_mass_transfer_string()` in `common_functions.py`. Following the logic in this function, the mass transfer case will always be assigned to `S1` and never assigned to `S2`. This is indeed the case in the output binary populations, as all mass transfer cases end in `1`.
Finally, `mass_transfer_case` is not included in the `flip_stars()` function at all in `common_functions.py`. 
The simplest fix to the above 2 problems would be to remove the `1` or `2` indicators from the `mass_transfer_case` strings, as a user could determine which object is transferring mass from the `RLO1` or `RLO2` states.

To reproduce this behavior for a single binary, you can run the following code in a notebook on Quest using the latest development branch, up to commit e9c0304:

```
ini_filepath = "/projects/b1119/cliotine/pulsar_dev/pulsar_pops/runs/dec_2023/dMd_0.025/small_runs/params.ini"
sim_kwargs = simprop_kwargs_from_ini(ini_filepath)
sim_prop = SimulationProperties(**sim_kwargs)
sim_prop.load_steps(verbose=True)

star_1 = SingleStar(**{'mass': 6.769807, 'state': 'H-rich_Core_H_burning'})
star_2 = SingleStar(**{'mass' : 7.3149, 'state' : 'H-rich_Core_H_burning',\
                      'natal_kick_array': [52.375966, 3.971669, 1.969153, 6.103875]})
binary = BinaryStar(star_1, star_2,
                    **{'time': 0.0, 'state': 'detached', 'event': 'ZAMS', 'orbital_period': 79.939247, 'eccentricity': 0.0},
                    properties = sim_prop)

binary.evolve()
binary.to_df()
```

**Comment by celiotine** [comment]
Note: this is also happening for systems where `S2` is initially more massive than `S1`, so it could be related to issue #203 . However, when these binaries are re-run in a notebook like in the example above, the `S1` and `S2` objects flip according to the larger mass, but the labels in the `state` and `event` columns also flip and remain incorrect.

**Comment by celiotine** [comment]
Part of this problem is being cause by a bug in `step_merged.py`. In this step, the merged object and the massless remnant are returned, and in most cases, the merged object is the "base" object that engulfed its companion. However, in the special case where a star is engulfing a compact object, we want to keep the compact object the same and set its companion to massless remnant. Currently, the CO object is being flipped to the other star object.
After fixing this issue and testing for a single binary, the problem is resolved:
<img width="721" alt="Screen Shot 2023-12-13 at 3 01 52 PM" src="https://github.com/POSYDON-code/POSYDON/assets/13874913/342e8ac1-aae3-4840-a37c-a1185db938bd">

(This is the same binary I ran above)
I am currently testing this fix on a small population.

**Comment by celiotine** [comment]
Confirming that this fix performs well in a small population. It also fixes the issue of S2 originally being heavier than S1 (issue #203) for a majority of binaries. Will submit a PR.

**Comment by ezapartas** [comment]
@dimsour94

**Comment by mkruckow** [comment]
Should be fixed with PR #219 .
